ftw/calendarexport/export.py

```diff
@@ -24,17 +25,29 @@ class ExportICS(CalendarView):
     def update(self):
         context = aq_inner(self.context)
         catalog = getToolByName(context, 'portal_catalog')
-        provides = ICalendarSupport.__identifier__
         uids = self.request.form.get('uids', [])
-        self.events = catalog(UID=uids, object_provides=provides)
+        self.events = catalog(UID=uids)
         if not uids:
             self.events = []
 
     @ram.cache(cachekey)
     def feeddata(self):
         data = cs.ICS_HEADER % dict(prodid=cs.PRODID)
         for brain in self.events:
-            tmp_data = brain.getObject().getICal()
-            data += tmp_data
+            obj = brain.getObject()
+            event_data = ""
+
+            if ICalendarSupport.providedBy(obj):
+                event_data = obj.getICal()
+
+            if PLONE_APP_EVENTS_AVAILABLE:
+                from plone.app.event.dx.interfaces import IDXEvent
+                from plone.app.event.ical import construct_icalendar
```

Comment:
Since these imports are within the `for`-loop, they are probably executed in each cycle.
Maybe moving the imports to the module level (within an `if`) could make it a little faster. See [Import Statement Overhead](https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Import_Statement_Overhead).